cbor/cbor_test.go

```diff
@@ -2103,3 +2104,51 @@ func FuzzUnmarshal(f *testing.F) {
 		_ = cbor.Unmarshal(data, &v)
 	})
 }
+
+func TestEmbeddedStruct(t *testing.T) {
+
+	// Test Setup
+	type E struct {
+		B string
+		C []byte
+		D int
+	}
+	type st struct {
+		A int
+		E
+	}
+	c := make([]byte, 0, 4)
+	c = append(c, 12, 13, 14, 15)
+	input := st{A: 1, E: E{B: "test", C: c, D: 15}}
+
+	//Perform marshalling and unmarshalling of data
+	byteData, err := cbor.Marshal(input)
+	if err != nil {
+		fmt.Errorf("error in Marshalling data:%w", err)
+	}
+
+	var v st
+	if err = cbor.Unmarshal(byteData, &v); err != nil {
+		fmt.Errorf("error in unmarshalling data:%w", err)
```

Comment:
The two `fmt.Errorf` need to be `t.Fatalf`.

But instead, can you just make the existing embedded struct encode and decode tests fail for this case?

Reply:
Specifically `TestEncodeArray/tuples` and `TestDecodeArray/tuples`.

Reply:
And thanks for all the work uncovering this bug! If you don't have time to make these test changes, just let me know and I'll make a commit co-authored by both of us.

Reply:
I have covered embedded struct, array and slice in embedded struct in a test case. Can you please check?
I can take a look at it in next week again. Please feel free to contribute to this PR, if required. Thank you.